internal/plugin/connectors/tcp/mssql/connection_details.go

```diff
@@ -39,6 +40,12 @@ func NewConnectionDetails(credentials map[string][]byte) *ConnectionDetails {
 		connDetails.Password = string(credentials["password"])
 	}
 
+	if credentials["sslmode"] == nil {
+		connDetails.SSLMode = "disable"
+	} else {
+		connDetails.SSLMode = string(credentials["sslmode"])
```

Comment:
Maybe this is related to the test comments below, but don't we currently only support "disable"? This feels like a reasonable spot to throw an error if a different value is set here

Reply:
I'm still unclear as to how we'd want to throw errors here, as we only have infrastructure for writing fake mssql errors to the client, and we have avoided throwing secretless specific errors. We discussed adding error handling to this exact method in the past, and decided against it at the time. I'm not against it, I just think it's out of scope for this story. 

For now, I set a switch that only accepts "disable", and any other value falls through to the default, which is "disable" currently as well.